src/os/bluestore/BlueFS.cc

```diff
@@ -3086,6 +3108,18 @@ void BlueFS::flush_range(FileWriter *h, uint64_t offset, uint64_t length)/*_WF*/
   _maybe_check_vselector_LNF();
   std::unique_lock hl(h->lock);
   _flush_range_F(h, offset, length);
+
+  // For bluefs_buffered_io=false, data is submitted via libaio
+  // we need to make sure aio is returned before updating the fnode's size
+  _flush_bdev(h);
+
+  // reset the fnode's inflight_flush_size since they are on disk now.
+  // if compact_log is triggered before this step, it dumps this fnode
+  // with size that excludes inflight_flush_size which is fine, since flush()
+  // doesn't guarantee metadata is updated on this file and we didn't break the
+  // semantics. if compact_log is triggered after this step, it dumps fnode contains
+  // the newly flushed data which is also accurate.
+  h->file->inflight_flush_size = 0;
```

Comment:
this is accessed out of file->lock - not sure this is valid. May be make inflight_flush_size atomic?

Reply:
If BlueFS in ceph is only invoked via RocksDB layer, then it won't be a problem. Since RocksDB never open the same file for write across multiple threads concurrently.